packages/graphile-build-pg/src/plugins/PgIntrospectionPlugin.js

```diff
@@ -139,287 +138,377 @@ function readFile(filename, encoding) {
-      "id"
-    );
-
-
-    /**
-     *
-     *
-     * @param array
-     * @param newAttr
-     * @param lookupAttr
-     * @param lookup
-     * @param missingOk
-     */
-    const relate = (array, newAttr, lookupAttr, lookup, missingOk = false) => {
-      array.forEach(entry => {
-
-        //relationship
-        const key = entry[lookupAttr];
-
-        if (Array.isArray(key)) {
-          entry[newAttr] = key
-            .map(innerKey => {
-              const result = lookup[innerKey];
-              if (innerKey && !result) {
-                if (missingOk) {
-                  return;
-                }
-                throw new Error(
-                  `Could not look up '${newAttr}' by '${lookupAttr}' ('${innerKey}') on '${JSON.stringify(
-                    entry
-                  )}'`
-                );
-              }
-              return result;
-            })
-            .filter(_ => _);
-        } else {
-          const result = lookup[key];
-          if (key && !result) {
-            if (missingOk) {
-              return;
-            }
-            throw new Error(
-              `Could not look up '${newAttr}' by '${lookupAttr}' on '${JSON.stringify(
-                entry
-              )}'`
-            );
-          }
-          entry[newAttr] = result;
-        }
-      });
-    };
-
-    relate(
-      introspectionResultsByKind.class,
-      "namespace",
-      "namespaceId",
-      introspectionResultsByKind.namespaceById,
-      true // Because it could be a type defined in a different namespace - which is fine so long as we don't allow querying it directly
-    );
-
-    relate(
-      introspectionResultsByKind.class,
-      "type",
-      "typeId",
-      introspectionResultsByKind.typeById
-    );
-
-    relate(
-      introspectionResultsByKind.attribute,
-      "class",
-      "classId",
-      introspectionResultsByKind.classById
-    );
-
-    relate(
-      introspectionResultsByKind.attribute,
-      "type",
-      "typeId",
-      introspectionResultsByKind.typeById
+      },
+      {}
     );
+  });
+  pgStructuresByKind.attributeByClassIdAndNum = pgStructuresByKind.attribute.reduce(
+    (memo, struct) => {
+      const { classId, num } = struct;
+      if (!memo.hasOwnProperty(classId)) memo[classId] = {};
+      memo[classId][num] = struct;
+      return memo;
+    },
+    {}
+  );
+
+  //TODO if yes, can warn about the deprecation via proxies (example below)
+  // ["namespaceById", "classById", "typeById", "extensionById"].forEach(prop => {
+  //     console.log(prop);
+  //    pgStructuresByKind[prop] = new Proxy(pgStructuresByKind[prop], {
+  //        get: function(target, property, rec){
+  //            console.warn(`@DeprecationWarning: Accessing postgres data by ID via pgIntrospectionResultsByKind will
+  //            soon be deprecated. Please use pgIntrospectionResultsByID instead.`);
+  //            return target[property];
+  //        }
+  //    })
+  // });
```

Comment:
Lets skip this for now, I'm happy with the current API and would rather not maintain two interfaces.

Reply:
Absolutely. I got a little ahead of myself. I think looking up the introspection results by a globally unique identifier would simplify the api a bit, but given the comments about the oid uniqueness, this should be put on the backburner. I'll remove my changes here from the PR as soon as we conclude on the best way to handle oid logic.